js/iframe-coordinator/Elm.d.ts

```diff
@@ -1,69 +1,58 @@
-declare module "*/Client.elm" {
-  export interface LabeledMsg {
-    msgType: string;
-    msg: any;
-  }
+interface LabeledMsg {
+  msgType: string;
+  msg: any;
+}
 
-  export interface MessageData {
-    origin: string;
-    data: any;
-  }
+interface SubscribeHandler {
+  (msg: LabeledMsg): void;
+}
 
-  export interface Worker {
-    requestToast: Function;
-    ports: {
-      fromHost: {
-        send: (message: MessageData) => void;
-      };
-      fromClient: {
-        send: (message: LabeledMsg) => void;
-      };
-      toHost: {
-        subscribe: (message: any) => void;
-      };
-      toClient: {
-        subscribe: (message: any) => void;
-      };
-    };
-  }
+interface InPort {
+  send(msg: LabeledMsg): void;
+}
 
-  interface Client {
-    worker(): Worker;
-  }
+interface OutPort {
+  subscribe(handler: SubscribeHandler): void;
+}
 
-  export var Client: Client;
+interface ClientProgram {
+  ports: {
+    fromHost: {
+      send({ origin: string, data: any });
+    };
+    toHost: OutPort;
+    fromClient: InPort;
+    toClient: OutPort;
+  };
 }
 
-declare module "*/Host.elm" {
-  export interface ClientRegistrations {}
+declare module "*/Client.elm" {
+  export var Elm: {
```

Comment:
do we need something more unique here?

Reply:
In 0.19 Elm exports everything under a single `Elm` root object. So the uniqueness comes from the combined `Elm.Client`. If our Elm project exposed a `Foo` module, it'd be at `Elm.Foo`

Reply:
It's a bit weird, because in webpack, we're including things at a file level, which makes importing `Elm.Client` from `Client.elm` a little bit redundant. But you could compile the Elm in other ways that led to multiple files & modules in a single bundle.